net/node/transactionPool.go

```diff
@@ -14,88 +16,365 @@ import (
 
 type TXNPool struct {
 	sync.RWMutex
-	txnCnt uint64
-	list   map[common.Uint256]*transaction.Transaction
+	txProcessList     // transaction which have been verifyed will put into this map
+	tXNPendingList    // transaction which didn't pass the verify will put into this map
+	assetIssueSummary // transaction which pass the verify will summary the amout to this map
+	inputUTXOList     // transaction which pass the verify will add the UTXO to this map
 }
 
-func (txnPool *TXNPool) GetTransaction(hash common.Uint256) *transaction.Transaction {
-	txnPool.RLock()
-	defer txnPool.RUnlock()
-	txn := txnPool.list[hash]
-	// Fixme need lock
-	return txn
+func (txnPool *TXNPool) init() {
+	txnPool.initAssetIssueSummary()
+	txnPool.initTxProcessList()
+	txnPool.initTXNPendingList()
+	txnPool.initInputUTXOList()
 }
 
 func (txnPool *TXNPool) AppendTxnPool(txn *transaction.Transaction) bool {
-	hash := txn.Hash()
-	// TODO: Call VerifyTransactionWithTxPool to verify tx
+	//verify transaction with Concurrency
+	if err, errCode := va.VerifyTransactionCanConcurrency(txn); err != nil {
+		if errCode != errors.ErrDuplicatedTx {
+			txnPool.appendTXNPendingList(txn, errCode)
+		}
+		return false
+	}
+
+	//verify transaction by pool with lock
 	txnPool.Lock()
-	txnPool.list[hash] = txn
-	txnPool.txnCnt++
-	txnPool.Unlock()
+	defer txnPool.Unlock()
+	if ok := txnPool.verifyTransactionWithTxPool(txn); !ok {
+		return false
+	}
+	txnPool.appendToProcessList(txn)
 	return true
 }
 
 // Attention: clean the trasaction Pool after the consensus confirmed all of the transcation
 func (txnPool *TXNPool) GetTxnPool(cleanPool bool) map[common.Uint256]*transaction.Transaction {
 	txnPool.Lock()
 	defer txnPool.Unlock()
+	txList := txnPool.getProcessTxnList(cleanPool)
+	return txList
+}
 
-	list := txnPool.list
-	if cleanPool == true {
-		txnPool.init()
+//Attention: clean the trasaction Pool with committed transactions.
+func (txnPool *TXNPool) CleanSubmittedTransactions(block *ledger.Block) error {
```

Comment:
The parameter here should be the transaction list instead of the the block,  what's your opinion